### PR TITLE
Fix allocator sizeof operand mismatch detected by clang CI

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/io/p/io_winedbg.c [hunk 1]
```diff
@@ -243,7 +243,7 @@ static char *__system(RzIO *io, RzIODesc *fd, const char *cmd) {
 		eprintf("pid : show current process id\n");
 	} else if (!strncmp(cmd, "dr8", 3)) {
 		struct winedbg_x86_32 r = regState();
-		ut8 *arena = (ut8 *)calloc(3, sizeof(struct winedbg_x86_32));
+		struct winedbg_x86_32 *arena = calloc(3, sizeof(struct winedbg_x86_32));
```
Comment (ret2libc) [inline librz/io/p/io_winedbg.c]: can you use RZ_NEWS since you are changing this?

File: librz/util/regex/regcomp.c [hunk 2]
```diff
@@ -279,7 +279,7 @@ RZ_API int rz_regex_comp(RzRegex *preg, const char *pattern, int cflags) {
 		len = strlen((char *)pattern);
 	}
 	/* do the mallocs early so failure handling is easy */
-	g = (struct re_guts *)calloc(sizeof(struct re_guts) + (NC - 1), sizeof(cat_t));
+	g = calloc(1, (NC - 1) + sizeof(struct re_guts));
```
Comment (ret2libc) [inline librz/util/regex/regcomp.c]: RZ_NEW ?

Comment (ghost) [reply librz/util/regex/regcomp.c]: not possible because it is sizeof(struct re_guts) **+** (NC - 1)

Comment (ret2libc) [reply librz/util/regex/regcomp.c]: 👍 right, sorry

**Review by ret2libc — APPROVED** [review]

**Pull request merged**